apps/CMakeLists.txt

```diff
@@ -19,7 +19,11 @@ enable_testing()
 # add_subdirectory(HelloWasm)  # TODO(#5374): missing CMake build
 # add_subdirectory(HelloiOS)  # TODO(#5374): missing CMake build
 # add_subdirectory(auto_viz)  # TODO(#5374): missing CMake build
-add_subdirectory(bgu)
+if (Halide_ENABLE_ASAN)
```

Comment:
I think you meant `Halide_ASAN_ENABLED` here, but I think the approach is off.

A third-party user might add the asan flags themselves without using our toolchain file. We can detect whether this is the case by using `check_symbol_exists(__SANITIZE_ADDRESS__ "" ASAN_ENABLED)` and then branching off `ASAN_ENABLED`. All implementations that I'm aware of define `__SANITIZE_ADDRESS__` iff ASAN is enabled.

Reply:
> All implementations that I'm aware of define __SANITIZE_ADDRESS__ iff ASAN is enabled.

That may be, but it's not the way Clang documents. See Util.h for that.

Reply:
> A third-party user might add the asan flags themselves without using our toolchain file.

Not sure I want to care about that much, because I don't think that approach is gonna work properly (unless they add their own toolchain to ensure everything uses the same compiler). If it was that easy, that's what I would have done in the first place :-)

Reply:
I am actually concerned about divergence between CMake's and Halide's sources' view of whether or not ASAN is enabled. The following check should keep them synced:

```
include(CheckCXXSymbolExists)
check_cxx_symbol_exists(HALIDE_INTERNAL_USING_ASAN "${Halide_SOURCE_DIR}/src/Util.h" ASAN_ENABLED)
```